core/src/main/java/org/elasticsearch/common/Randomness.java

```diff
@@ -109,13 +113,42 @@ public static Random get() {
         }
     }
 
+    /**
+     * Provides a source of secure randomness that is reproducible when
+     * running under the Elasticsearch test suite, and otherwise
+     * produces a non-reproducible source of randomness. Reproducible
+     * sources of randomness are created when the system property
+     * "tests.seed" is set and the security policy allows reading this
+     * system property. Otherwise, non-reproducible sources of secure
+     * randomness are created.
+     *
+     * @return a source of randomness
+     * @throws IllegalStateException if running tests but was not able
+     *                               to acquire an instance of Random from
+     *                               RandomizedContext or tests are
+     *                               running but tests.seed is not set
+     */
+    public static Random getSecureRandom() {
```

Comment:
Maybe use `getSecure()` so that it's closer to the other methods in this class (`get()`).